sim/pokemon.ts

```diff
@@ -1234,10 +1234,8 @@ export class Pokemon {
 			for (const typeid of type) {
 				if (this.hasType(typeid)) return true;
 			}
-		} else {
-			if (this.getTypes().includes(type)) {
-				return true;
-			}
+		} else if (this.getTypes().includes(type)) {
+			return true;
```

Comment:
This is also a situation where the original code is clearer: There's one check for the case where the types are an array, and another check for the case where the types aren't an array. It makes sense for them to be at the same level of indentation.

Reply:
make the else `return false`, then `return this.getTypes().includes(type)` outside of the block

Reply:
If I were writing this myself, I'd do something like:

```ts
	hasType(type: string | string[]) {
		if (!type) return false;
		if (typeof type === 'string')) {
			return this.getTypes().includes(type);
		}

		const types = this.getTypes();
		for (const typeName of type) {
			if (types.includes(typeName)) return true;
		}
		return false;
	}
```

Reply:
Sure, go with that